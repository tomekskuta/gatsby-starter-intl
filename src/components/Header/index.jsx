import React from 'react';
import PropTypes from 'prop-types';
import Link from '../Link';
import Langs from '../Langs';

const Header = ({ siteTitle, hideLangs }) => (
  <header
    style={{
      background: `rebeccapurple`,
      marginBottom: `1.45rem`,
    }}
  >
    <div
      style={{
        margin: `0 auto`,
        maxWidth: 960,
        padding: `1.45rem 1.0875rem`,
        display: 'flex',
        justifyContent: 'space-between',
        flexWrap: 'wrap',
      }}
    >
      <h1 style={{ margin: 0 }}>
        <Link
          to="/"
          style={{
            color: `white`,
            textDecoration: `none`,
          }}
        >
          {siteTitle}
        </Link>
      </h1>
      {!hideLangs && <Langs />}
    </div>
  </header>
);

Header.propTypes = {
  siteTitle: PropTypes.string,
  hideLangs: PropTypes.bool,
};

Header.defaultProps = {
  siteTitle: ``,
  hideLangs: false,
};

export default Header;
