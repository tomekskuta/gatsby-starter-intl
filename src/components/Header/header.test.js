import React from 'react';
import renderer from 'react-test-renderer';

import Header from './index';

describe('Header', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Header siteTitle="Default Starter" hideLangs={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
