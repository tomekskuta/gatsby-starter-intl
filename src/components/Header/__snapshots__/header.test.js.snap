// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  style={
    Object {
      "background": "rebeccapurple",
      "marginBottom": "1.45rem",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flexWrap": "wrap",
        "justifyContent": "space-between",
        "margin": "0 auto",
        "maxWidth": 960,
        "padding": "1.45rem 1.0875rem",
      }
    }
  >
    <h1
      style={
        Object {
          "margin": 0,
        }
      }
    >
      <a
        href="/undefined/"
        style={
          Object {
            "color": "white",
            "textDecoration": "none",
          }
        }
      >
        Default Starter
      </a>
    </h1>
  </div>
</header>
`;
